muxrpc/handlers/alias/handler.go

```diff
@@ -147,3 +147,47 @@ func (h Handler) Revoke(ctx context.Context, req *muxrpc.Request) (interface{},
 
 	return true, nil
 }
+
+func (h Handler) List(ctx context.Context, req *muxrpc.Request) (interface{}, error) {
+	var args []string
+
+	err := json.Unmarshal(req.RawArgs, &args)
+	if err != nil {
+		return nil, fmt.Errorf("listAlias: bad request: %w", err)
+	}
+
+	if n := len(args); n != 1 {
+		return nil, fmt.Errorf("listAlias: expected one argument got %d", n)
+	}
+
+	ref, err := refs.ParseFeedRef(args[0])
+	if err != nil {
+		return nil, fmt.Errorf("listAlias: invalid feed ref: %w", err)
+	}
+
+	allAliases, err := h.db.List(ctx)
+	if err != nil {
+		var takenErr roomdb.ErrAliasTaken
+		if errors.As(err, &takenErr) {
+			return nil, takenErr
+		}
+		return nil, fmt.Errorf("listAlias: could not register alias: %w", err)
```

Comment:
We're not trying to register, we're trying to list?

Reply:
Right, this error handling in general was just bollocks, I removed it. Sorry, I whipped this out pretty quickly. There are some mistakes like this with error strings in other handlers as well which I will fix in a separate PR.

Reply:
6b932f2

Reply:
Nice one, thanks!